### PR TITLE
NSA_2899 DQT report generation

### PR DESCRIPTION
**Pull request by arshedmust**

### CODE REVIEW

File: src/controllers/dqtReport.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
+import { Request, Response, NextFunction } from "express";
+import { Controller, Get, Post, Middleware } from "@overnightjs/core";
+import fs from "fs";
+import _ from "lodash";
```
Comment (DrogoNevets) [inline src/controllers/dqtReport.ts]: lets only import the functions we are using not the whole of lodash

File: src/controllers/dqtReport.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    return;
+  }
+
+  @Post("/")
+  async post(req: Request, res: Response): Promise<void> {
+    try {
+      const startDate = req.body.startdate;
+      const endDate = req.body.enddate;
+      //Todo: Form validation needed
+      console.log(startDate + " " + endDate);
```
Comment (DrogoNevets) [inline src/controllers/dqtReport.ts]: lets drop this

File: src/controllers/dqtReport.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      await dbAuditSvc.connect();
```
Comment (DrogoNevets) [inline src/controllers/dqtReport.ts]: this should be done in the service layer not controller

File: src/controllers/dqtReport.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+      let stat,
+        file,
+        signins = await dbAuditSvc.findAllSignIns(startDate, endDate);
```
Comment (DrogoNevets) [inline src/controllers/dqtReport.ts]: connect should be part of this call

File: src/controllers/dqtReport.ts [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      dbAuditSvc.destroy();
```
Comment (DrogoNevets) [inline src/controllers/dqtReport.ts]: as should this

File: src/controllers/dqtReport.ts (hunk 1 above)
Comment (arshedmust) [reply src/controllers/dqtReport.ts]: Done

File: src/controllers/dqtReport.ts (hunk 2 above)
Comment (arshedmust) [reply src/controllers/dqtReport.ts]: Assuming you mean the console logs ? So dropped them

File: src/controllers/dqtReport.ts (hunk 3 above)
Comment (arshedmust) [reply src/controllers/dqtReport.ts]: Moved to service layer

File: src/controllers/dqtReport.ts (hunk 4 above)
Comment (arshedmust) [reply src/controllers/dqtReport.ts]: Done

File: src/controllers/dqtReport.ts (hunk 5 above)
Comment (arshedmust) [reply src/controllers/dqtReport.ts]: Done

**Pull request merged**